custom_dc/load_data.sh

```diff
@@ -0,0 +1,319 @@
+  [[ -n "$CONFIG" ]] && importer_options="$importer_options --config_file=$CONFIG"
+  [[ -n "$MODE" ]] && importer_options="$importer_options --mode=$MODE"
+  [[ -n "$OUTPUT_DIR" ]] && importer_options="$importer_options --output_dir=$OUTPUT_DIR"
+  mkdir -p $OUTPUT_DIR
+  OUTPUT_DIR=$(readlink -f $OUTPUT_DIR)
+
+  # Clear state from old run
+  report_json=$OUTPUT_DIR/process/report.json
+  rm $report_json
+
+  # Run the simple importer
+  local cwd="$PWD"
+  cd "$SIMPLE_DIR"
+  cmd="python -m stats.main $importer_options"
+  echo_log "Running command: $cmd"
+  $cmd >> $LOG 2>&1
+  cmd_status=$?
+  cd "$cwd"
+  if [[ "$cmd_status" != "0" ]]; then
+    echo_fatal "Failed to run simple importer: $cmd. Logs in $LOG"
+  fi
+
+  status=$(grep '"status"' $report_json)
+  echo_log "simple importer: $status"
+}
+
+# Generate tmcf for a csv file in dir
+function get_or_generate_tmcf {
+  set -x
+  local dir="$1"; shift
+
+  tmcf=$(ls $dir/*.tmcf | head -1)
+  [[ -n "$tmcf" ]] && echo "$tmcf" && return
+
+  # Create a new tmcf for columns in csv
+  csv=$(ls $dir/*.csv | head -1)
+  tmcf=$(echo "$csv" | sed -e 's/.csv$/.tmcf/')
+  # TODO: generate tmcf based on csv columns
+  cat <<END_TMCF > $tmcf
+Node: E:Stats->E0
+typeOf: dcs:StatVarObservation
+observationAbout: C:Stats->entity
+observationDate: C:Stats->date
+variableMeasured: C:Stats->variable
+value: C:Stats->value
+END_TMCF
+  echo "$tmcf"
+  set +x
+}
+
+# Run dc-import genmcf to validate generated csv/mcf files.
+function validate_output {
+  #TODO: For customdc validate data in sql triples db and observations db
+  # get output csv if any.
+  local output_csv=$(ls $OUTPUT_DIR/*.csv 2>/dev/null)
+  if [[ -z "$output_csv" ]]; then
+    echo_log "Skipping validate as there are no csv files in $OUTPUT_DIR"
+    return
+  fi
+  setup_dc_import
+  echo_log "Validating output in $OUTPUT_DIR"
+
+  # Get the tmcf file to validate csv stats.
+  tmcf=$(get_or_generate_tmcf "$OUTPUT_DIR")
+
+  # Run dc-import
+  DC_IMPORT_CMD="genmcf"
+  [[ "$MODE" == "customdc" ]] && DC_IMPORT_CMD="lint"
+  cmd="java -jar $DC_IMPORT_JAR $DC_IMPORT_CMD -n 20 -r FULL $OUTPUT_DIR/*.csv $tmcf -o $OUTPUT_DIR/dc_generated"
+  run_cmd $cmd
+  echo_log "Output of validation in $OUTPUT_DIR/dc_generated/report.json"
+}
+
+# Generate embeddings for sentences
+function generate_embeddings {
+  setup_python
+  EMBEDDINGS_TOOL_DIR=${EMBEDDINGS_TOOL_DIR:-"$WEBSITE_DIR/tools/nl/embeddings"}
+  NL_DIR=${NL_DIR:-"$OUTPUT_DIR/nl"}
+  echo_log "Building embeddings for sentences in $NL_DIR"
+  local cwd="$PWD"
+  cd $EMBEDDINGS_TOOL_DIR
+  run_cmd python3 build_custom_dc_embeddings.py --sv_sentences_csv_path=$NL_DIR/sentences.csv --output_dir=$NL_DIR
+  cd "$cwd"
+}
+
+# Copy files from local folder to GCS
+function copy_to_gcs {
+  local dir="$1"; shift
+  local gcs_dir="$1"; shift
+
+  [[ -z "$gcs_dir" ]] && return
+
+  cmd="gsutil -m cp -r $dir $gcs_dir"
```

Comment:
This will require them to have `gsutil` installed. I'm wondering whether we want that or have the RSI script handle GCS uploads by itself?

Reply:
Can the build_embeddings script also work with GCS folders? 

Then we can skip the GCS copy as it is only used for the dc-import validation step not used here.

Reply:
verified that build embeddings works with GCS OUTPUT_DIR. removed copy stage.